engine/src/main/java/io/zeebe/engine/processor/TypedRecord.java

```diff
@@ -22,8 +22,12 @@
 
 public interface TypedRecord<T extends UnpackedObject> {
 
+  long getPosition();
+
   long getKey();
 
+  long getSourceEventPosition();
```

Comment:
I don't think this method should be in the interface. It is not always implement, i.e. `TypedWorkflowInstanceRecord` always returns 0. (https://github.com/zeebe-io/zeebe/pull/2491/files#diff-37ffb731f1b5ec8f32004d4b5ca882ecR561)

Also it is only used in test assertions and filters on record streams, where we also have access to the full record.